src/TSID/src/QPFixedBaseTSID.cpp

```diff
@@ -0,0 +1,601 @@
+/**
+ * @file QPFixedBaseTSID.cpp
+ * @authors Ines Sorrentino
+ * @copyright 2021 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <BipedalLocomotion/TSID/QPFixedBaseTSID.h>
+#include <BipedalLocomotion/TextLogging/Logger.h>
+
+#include <OsqpEigen/OsqpEigen.h>
+
+#include <BipedalLocomotion/TSID/SE3Task.h>
+#include <BipedalLocomotion/TSID/JointDynamicsTask.h>
+
+#include <BipedalLocomotion/ParametersHandler/StdImplementation.h>
+
+using namespace BipedalLocomotion;
+using namespace BipedalLocomotion::TSID;
+using namespace BipedalLocomotion::ParametersHandler;
+
+struct QPFixedBaseTSID::Impl
+{
+    struct TaskWithPriority
+    {
+        std::shared_ptr<System::LinearTask> task;
+        std::size_t priority;
+        Eigen::VectorXd weight;
+        Eigen::MatrixXd tmp; /**< This is a temporary matrix usefull to reduce dynamics allocation
+                                in advance method */
+    };
+    
+    std::unordered_map<std::string, TaskWithPriority> tasks;
+    std::vector<std::reference_wrapper<const TaskWithPriority>> constraints;
+    std::vector<std::reference_wrapper<TaskWithPriority>> costs;
```

Comment:
Could you add a bit of a documentation here? Also why both constraints and costs have `TaskWithPriority` type? These are the tasks that are used to build the cost function.